BetterThanRottenTomatoAPI/ReviewManagement/ReviewManagement.API/Controllers/ReviewController.cs

```diff
@@ -0,0 +1,76 @@
+﻿using Microsoft.AspNetCore.Mvc;
+using ReviewManagement.API.Models;
+using ReviewManagement.Domain.Interfaces;
+using ReviewManagement.Domain.Models;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace ReviewManagement.API.Controllers
+{
+    [Route("api/reviews")]
+    [ApiController]
+    public class ReviewController : ControllerBase
+    {
+        private IReviewService _reviewService;
+
+        public ReviewController(IReviewService reviewService)
+        {
+            _reviewService = reviewService;
+        }
+
+        [HttpPost]
+        [Route("addreview")]
+        public async Task<IActionResult> AddMovieReview([FromBody] AddMovieReviewRequest movieReview)
+        {
+            if(movieReview == null)
+            {
+                return StatusCode(400);
+            }
+            try
+            {
+                MovieReviewModel domainMovieReview = new MovieReviewModel()
+                {
+                    Username = movieReview.Username,
+                    Comment = movieReview.Comment,
+                    MovieId = movieReview.MovieId,
+                    Rating = movieReview.Rating
+                };
+                await _reviewService.AddMovieReviewService(domainMovieReview);
+                return StatusCode(201);
+            }
+            catch(ArgumentException ae)
+            {
+                return StatusCode(400, ae.Message);
+            }
+            catch(Exception e)
+            {
+                return StatusCode(500, e.Message);
+            }
+        }
+        [HttpGet]
+        [Route("getmoviereviews/{movieId}")]
+        public async Task<IActionResult> GetMovieReviews(long movieId, [FromBody] int page)
```

Comment:
[FromQuery] not [FromBody].

We only use [FromBody] if you are actually sending a big body to GET request that can't be put in the URL. Since its only the page, we can put that in the query parameter of url so its [FromQuery]